packages/grid/src/Grid.tsx

```diff
@@ -1,81 +1,313 @@
+          ) : null }
+
+          <BodyCellContent
+            theme={ themeRef.current }
+            expandLevel={ expandLevel }
+            center={ !!column.center }
+            selected={ isSelected }
+          >
+            <span>{ content }</span>
+          </BodyCellContent>
+        </BodyCell>
+      </CellMeasurer>
+    );
+  };
+
+  const handleChangeWidth = useCallback(
+    (index, newWidth) => {
+      const newColumns = setIn(mappedColumns, newWidth, [index, 'width']);
+      changeMappedColumns(newColumns);
+      fullWidth.current = newColumns.reduce(
+        (acc: number, { width: colWidth }: IMappedItem) => (+acc + +colWidth),
+        0,
+      );
+      onChangeColumns(newColumns);
+    },
+    [mappedColumns, onChangeColumns],
+  );
+
+
+  const handleChangeMoving = useCallback(
+    (newColumns) => {
+      changeMappedColumns(newColumns);
+      onChangeColumns(newColumns);
+    },
+    [onChangeColumns],
+  );
 
   useEffect(() => {
-    themeRef.current = gridTheme<ITableTheme>(theme);
-  }, [theme]);
+    if (gridRef.current) gridRef?.current?.recomputeGridSize();
+    if (cache.current) cache.current.clearAll();
+  }, [mappedColumns, mappedRows]);
+
+  const {
+    header: { height: headerHeight = 0 },
+    totals: { height: totalsHeight = 0 },
+  } = themeRef.current;
 
   return (
-    <ContentWrapper theme={ themeRef.current }>
-      <HeadWrapper>
-        <TableHead style={ { transform: `translateX(-${tableScroll}px)` } } theme={ themeRef.current }>
-          { columns.map(({ title, width }: IColumn) => (
-            <HeaderStyled
-              width={ width }
-              key={ title }
-              theme={ themeRef.current.head }
-            >
-              { title }
-            </HeaderStyled>
-          )) }
-        </TableHead>
-      </HeadWrapper>
-      <TableWrapper
-        ref={ tableWrapperElement }
-        onScroll={ handleScroll }
-      >
-        <TableStyled>
-          { rows.map((row: IRowData) => (
-            <Row
-              row={ row }
-              key={ row.id }
-              columns={ columns }
+    <Wrapper theme={ themeRef.current } width={ width } changingColumns={ changingColumns }>
+      <HeaderWrapper
+        fullWidth={ fullWidth.current }
+        columns={ mappedColumns }
+        translateX={ scrollLeft }
+        onChangeWidth={ handleChangeWidth }
+        onChangeMoving={ handleChangeMoving }
+        changeChangingColumns={ changeChangingColumns }
+        theme={ themeRef.current }
+        shouldMovingColumns={ shouldMovingColumns }
+        shouldChangeColumnsWidth={ shouldChangeColumnsWidth }
+      />
+      <Body>
+        <VirtualisedGrid
+          ref={ gridRef as MutableRefObject<VirtualisedGrid> }
+          columnCount={ mappedColumns.length }
+          columnWidth={ ({ index }: any) => mappedColumns[index].width }
+          deferredMeasurementCache={ cache.current }
+          height={ height - +headerHeight - +totalsHeight }
```

Comment:
Why there is + before headerHeight and totalsHeight?
Is there are cases when they are negative?